pkg/common/utils/utils.go

```diff
@@ -257,3 +259,109 @@ func GetUserFromPod(pod *v1.Pod) string {
 
 	return value
 }
+
+func GetPodAnnotationValue(pod *v1.Pod, annotationKey string) string {
+	if value, ok := pod.Annotations[annotationKey]; ok {
+		return value
+	}
+	return ""
+}
+
+func GetNameSpaceAnnotationValue(namespace *v1.Namespace, annotationKey string) string {
+	if value, ok := namespace.Annotations[annotationKey]; ok {
+		return value
+	}
+	return ""
+}
+
+func GetPodLabelValue(pod *v1.Pod, labelKey string) string {
+	if value, ok := pod.Labels[labelKey]; ok {
+		return value
+	}
+	return ""
+}
+
+func GetTaskGroupFromPodSpec(pod *v1.Pod) string {
+	return GetPodAnnotationValue(pod, constants.AnnotationTaskGroupName)
+}
+
+func GetPlaceholderFlagFromPodSpec(pod *v1.Pod) bool {
```

Comment:
This function is not covered by unit tests. As I can see, it was missing in the original code.
Please write one to enhance coverage.

Reply:
Addressed in latest PR, thanks!